src/routes/auth/providers/utils.ts

```diff
@@ -84,3 +81,58 @@ export const providerCallback = async (req: Request, res: Response): Promise<voi
   // redirect back user to app url
   res.redirect(PROVIDERS_SUCCESS_REDIRECT as string)
 }
+
+interface Constructable<T> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  new (...args: any[]): T
+  prototype: T
+}
+
+const config = (_config as unknown) as Record<string, boolean | string>
+
+export const initProvider = <T extends Strategy>(
+  router: Router,
+  strategyName: string,
+  strategy: Constructable<T>,
+  options: ConstructorParameters<Constructable<T>>[0] // TODO: Strategy option type is not inferred correctly
+): void => {
+  // Strategy name in capitals, to get the configuration
+  const capsStragegy = strategyName.toUpperCase()
+  // Checks if the strategy is enabled. Don't create any route otherwise
+  if (config[`AUTH_${capsStragegy}_ENABLE`] as boolean) {
+    const clientID = config[`AUTH_${capsStragegy}_CLIENT_ID`]
+    const clientSecret = config[`AUTH_${capsStragegy}_CLIENT_SECRET`]
+    // Checks if the strategy has at least a client ID and a client secret
+    if (!clientID || !clientSecret) {
+      throw Boom.badImplementation(`Missing environment variables for ${strategyName} OAuth.`)
+    }
+
+    passport.use(
+      new strategy(
+        {
+          ...options,
+          clientID,
+          clientSecret,
+          callbackURL:
+            config[`AUTH_${capsStragegy}_CALLBACK_URL`] ||
+            `${SERVER_URL}/auth/providers/${strategyName}/callback`,
```

Comment:
I think we can safely assume that `${SERVER_URL}/auth/providers/${strategyName}/callback` should be enough. We should be able to omit the environment variable `AUTH_${capsStrategy}_CALLBACK_URL`